### ISSUE TITLE
DRY up location setting

### ISSUE THREAD

**Issue by aaronkavlie-wf** [Masterminds/glide#342]
The location is set mutliple times in the code base with a variation of the same code:

- [VcsUpdate](https://github.com/Masterminds/glide/blob/master/repo/vcs.go#L104-L109)
- [VcsGet cacheGoPath block](https://github.com/Masterminds/glide/blob/master/repo/vcs.go#L342-L347)
- [VcsGet cache block](https://github.com/Masterminds/glide/blob/master/repo/vcs.go#L373-L378)
- [VcsGet cache block again](https://github.com/Masterminds/glide/blob/master/repo/vcs.go#L397-L402)
- [2nd VcsGet cache block](https://github.com/Masterminds/glide/blob/master/repo/vcs.go#L448-L453)
- [Dependency.GetRepo](https://github.com/Masterminds/glide/blob/master/cfg/config.go#L428-L433)

this could be simplified with a small helper function.

**Comment by sdboyer** [comment]
yeah, it's a little bit of a warren in there. i'm rigorously refactoring in [vsolver](https://github.com/sdboyer/vsolver).
